pkg/controller/contrailmonitor/contrailmonitor_controller.go

```diff
@@ -0,0 +1,453 @@
+		IsController: true,
+		OwnerType:    &contrailv1alpha1.Contrailmonitor{},
+	})
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(&source.Kind{Type: &contrailv1alpha1.Postgres{}}, &handler.EnqueueRequestForOwner{
+		OwnerType: &contrailv1alpha1.Contrailmonitor{},
+	})
+	if err != nil {
+		return err
+	}
+	err = c.Watch(&source.Kind{Type: &contrailv1alpha1.Memcached{}}, &handler.EnqueueRequestForOwner{
+		OwnerType: &contrailv1alpha1.Contrailmonitor{},
+	})
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(&source.Kind{Type: &contrailv1alpha1.Cassandra{}}, &handler.EnqueueRequestForOwner{
+		OwnerType: &contrailv1alpha1.Contrailmonitor{},
+	})
+	if err != nil {
+		return err
+	}
+	err = c.Watch(&source.Kind{Type: &contrailv1alpha1.Control{}}, &handler.EnqueueRequestForOwner{
+		OwnerType: &contrailv1alpha1.Contrailmonitor{},
+	})
+	if err != nil {
+		return err
+	}
+	err = c.Watch(&source.Kind{Type: &contrailv1alpha1.Config{}}, &handler.EnqueueRequestForOwner{
+		OwnerType: &contrailv1alpha1.Contrailmonitor{},
+	})
+	if err != nil {
+		return err
+	}
+	err = c.Watch(&source.Kind{Type: &contrailv1alpha1.Keystone{}}, &handler.EnqueueRequestForOwner{
+		OwnerType: &contrailv1alpha1.Contrailmonitor{},
+	})
+	if err != nil {
+		return err
+	}
+	err = c.Watch(&source.Kind{Type: &contrailv1alpha1.Zookeeper{}}, &handler.EnqueueRequestForOwner{
+		OwnerType: &contrailv1alpha1.Contrailmonitor{},
+	})
+	if err != nil {
+		return err
+	}
+	err = c.Watch(&source.Kind{Type: &contrailv1alpha1.Rabbitmq{}}, &handler.EnqueueRequestForOwner{
+		OwnerType: &contrailv1alpha1.Contrailmonitor{},
+	})
+	if err != nil {
+		return err
+	}
+	err = c.Watch(&source.Kind{Type: &contrailv1alpha1.Webui{}}, &handler.EnqueueRequestForOwner{
+		OwnerType: &contrailv1alpha1.Contrailmonitor{},
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// blank assignment to verify that ReconcileContrailmonitor implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileContrailmonitor{}
+
+// ReconcileContrailmonitor reconciles a Contrailmonitor object
+type ReconcileContrailmonitor struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client     client.Client
+	scheme     *runtime.Scheme
+	manager    manager.Manager
+	kubernetes *k8s.Kubernetes
+}
+
+// Reconcile reads that state of the cluster for a Contrailmonitor object and makes changes based on the state read
+// and what is in the Contrailmonitor.Spec
+// TODO(user): Modify this Reconcile function to implement your Controller logic.  This example creates
+// a Pod as an example
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileContrailmonitor) Reconcile(request reconcile.Request) (reconcile.Result, error) {
```

Comment:
I don't see any unit tests for that controller. As far as I know there is a requirement that a new code should be covered by unit tests

Reply:
Moreover I strongly suggest to extend one of the e2e tests that will verify happy path. Just a single simple check

Reply:
I have not included unit test or e2e test cases here, I have noticed it, the reason is,  I get very inconstant results,  One day, I get one results, other day, i get completly diffrent results., I like you to suggest me what is wrong in the code that gives varied results.  for example,  even if the postgres service is active as it updates my additional print coloums crd instance it updates the field as Not active. it is very random

Reply:
If your code works inconsistently you should write unit tests and e2e tests and fix it. That is test driven development.